File: data/derived_cot/rq1_traces/translation/glm5.1/cpp_to_py/code_output/ChandrasekharSieve.py
class ChandrasekharSieve:
    def __init__(self, n: int):
        self.n = n
        self.primes = self._generate_primes()

    def _generate_primes(self) -> list[int]:
        if self.n < 2:
            return []

        sieve = [True] * (self.n + 1)
        sieve[0] = sieve[1] = False

        p = 2
        while p * p <= self.n:
            if sieve[p]:
                for i in range(p * p, self.n + 1, p):
                    sieve[i] = False
            p += 1

        return [i for i in range(2, self.n + 1) if sieve[i]]

    def get_primes(self) -> list[int]:
        return self.primes

File: data/derived_cot/rq1_traces/translation/glm5.1/java_to_cpp/code_output/ChandrasekharSieve.cpp
#include <iostream>
#include <vector>

class ChandrasekharSieve {
private:
    int n;
    std::vector<int> primes;

    std::vector<int> generatePrimes() {
        if (n < 2) {
            return std::vector<int>();
        }

        std::vector<bool> sieve(n + 1, true);
        sieve[0] = sieve[1] = false;

        int p = 2;
        while (p * p <= n) {
            if (sieve[p]) {
                for (int i = p * p; i <= n; i += p) {
                    sieve[i] = false;
                }
            }
            p += 1;
        }

        std::vector<int> result;
        for (int i = 2; i <= n; i++) {
            if (sieve[i]) {
                result.push_back(i);
            }
        }

        return result;
    }

public:
    ChandrasekharSieve(int n) : n(n), primes(generatePrimes()) {}

    std::vector<int> getPrimes() {
        return primes;
    }
};

int main() {
    ChandrasekharSieve cs(20);
    std::vector<int> primes = cs.getPrimes();

    std::cout << "[";
    for (size_t i = 0; i < primes.size(); i++) {
        if (i > 0) std::cout << ", ";
        std::cout << primes[i];
    }
    std::cout << "]" << std::endl;

    return 0;
}